src/modules/configs/config.py

```diff
@@ -59,88 +54,83 @@ class ApplicationConf(Conf):
+        self.application = ApplicationConf(
+            os.path.join(APPLICATION_DIR, "application.json")).parse_conf()
+        self.project = ProjectConf(os.path.join(APPLICATION_DIR, "project.json")).parse_conf()
+        self.operator = OperatorConf(os.path.join(APPLICATION_DIR, "operator.json")).parse_conf()
 
         self.merged_conf = {
-            "cli": self.merge_conf(examples.cli),
-            "application": self.merge_conf(examples.application),
-            "project": self.merge_conf(examples.project),
-            "operator": self.merge_conf(examples.operator)
+            "cli": {},
+            "application": self.merge_conf(self.application),
+            "project": self.merge_conf(self.project),
+            "operator": self.merge_conf(self.operator)
         }
 
-    # def create_final_conf(self):
-    #     final_conf = {}
-    #
-    #     for k, v in final.config.items():
-    #         for x, y in v.items():
-    #             # Repeat until lowest level
-    #             merged_key = f"{parent_key}{sep}{key}" if parent_key else key
-    #             rule = CONFIG_RULES.get(merged_key)
-    #             if rule:
-    #                 mode = rule.get("mode")
-    #                 if mode:
-    #                     order = rule.get(mode)
-    #                     if order:
-    #                         final_value = self.get_final_value(merged_key, mode, order)
-    #                         final[k][x] = final_value
-    #                     else:
-    #                         raise Exception("Order is undefined. Rule mode order needs to be defined for rule mode.")
-    #                 else:
-    #                     raise Exception("Mode is undefined")
-    #             else:
-    #                 raise Exception(f"No rules defined for conf key {merged_key}, update CONFIG_RULES.")
-    #
-    #     return final_conf
-    #
-    #
-    # def get_final_value(merged_key, mode, order):
-    #     value = self.apply_config_rule(merged_key, mode, order)
-    #     if value is None:
-    #         return CONFIG_RULES[merged_key].get("default")
-    #
-    #     return value
-    #
-    # def apply_config_rule(merged_key, mode, order):
-    #     value = None
-    #     if mode == "exclusive":
-    #         value = self.apply_exclusive_rule()
-    #     elif mode == "inclusive":
-    #         value = self.apply_inclusive_rule()
-    #     elif mode == "priority":
-    #         value = self.apply_priority_rule()
-    #     else:
-    #         raise Exception("Unexpected mode found, review CONFIG_RULES")
-    #
-    #     return value
-    #
-    # def apply_priority_rule(self, order):
-    #     """
-    #     Cascade/iterate conf by conf using defined order
-    #     If undefined at the end, return None
-    #     """
-    #     order =
-    #     for conf in order:
+        self.pre_final = self.create_final_conf()
+        self.final = self.validate_final_conf()
+
+    def validate_final_conf(self):
```

Comment:
This should be either inherited from the Conf class or the validation function should be extracted outside of Conf class (those will land in the `__init__` most likely but we will get to that later).